vm/src/obj/objlist.rs

```diff
@@ -181,6 +181,24 @@ fn list_add(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     }
 }
 
+fn list_iadd(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(
+        vm,
+        args,
+        required = [(zelf, Some(vm.ctx.list_type())), (other, None)]
+    );
+
+    if objtype::isinstance(other, &vm.ctx.list_type()) {
+        let mut elements = get_mut_elements(zelf);
+        for elem in get_elements(other).iter() {
+            elements.push(elem.clone());
```

Comment:
how about something like (I didn't test this)
```
elements.extend(get_elements(other).iter().cloned())
```
Simpler and may help performance, as `extend` can use iterator's size hint to preallocate all needed memory.